stdlib/public/Backtracing/Backtrace.swift

```diff
@@ -0,0 +1,496 @@
+    case omittedFrames(Int)
+
+    /// Indicates a discontinuity of unknown length.
+    case truncated
+
+    /// The program counter, without any adjustment.
+    public var originalProgramCounter: Address {
+      switch self {
+        case let .returnAddress(addr):
+          return addr
+        case let .programCounter(addr):
+          return addr
+        case let .asyncResumePoint(addr):
+          return addr
+        case .omittedFrames(_), .truncated:
+          return 0
+      }
+    }
+
+    /// The adjusted program counter to use for symbolication.
+    public var adjustedProgramCounter: Address {
+      switch self {
+        case let .returnAddress(addr):
+          return addr - 1
+        case let .programCounter(addr):
+          return addr
+        case let .asyncResumePoint(addr):
+          return addr
+        case .omittedFrames(_), .truncated:
+          return 0
+      }
+    }
+
+    /// A textual description of this frame.
+    public var description: String {
+      switch self {
+        case let .programCounter(addr):
+          return "\(hex(addr))"
+        case let .returnAddress(addr):
+          return "\(hex(addr)) [ra]"
+        case let .asyncResumePoint(addr):
+          return "\(hex(addr)) [async]"
+        case .omittedFrames(_), .truncated:
+          return "..."
+      }
+    }
+  }
+
+  /// Represents an image loaded in the process's address space
+  public struct Image: CustomStringConvertible, Sendable {
+    /// The name of the image (e.g. libswiftCore.dylib).
+    public var name: String
+
+    /// The full path to the image (e.g. /usr/lib/swift/libswiftCore.dylib).
+    public var path: String
+
+    /// The build ID of the image, as a byte array (note that the exact number
+    /// of bytes may vary, and that some images may not have a build ID).
+    public var buildID: [UInt8]?
+
+    /// The base address of the image.
+    public var baseAddress: Backtrace.Address
+
+    /// The end of the text segment in this image.
+    public var endOfText: Backtrace.Address
+
+    /// Provide a textual description of an Image.
+    public var description: String {
+      if let buildID = self.buildID {
+        return "\(hex(baseAddress))-\(hex(endOfText)) \(hex(buildID)) \(name) \(path)"
+      } else {
+        return "\(hex(baseAddress))-\(hex(endOfText)) <no build ID> \(name) \(path)"
+      }
+    }
+  }
+
+  /// A list of captured frame information.
+  public var frames: [Frame]
+
+  /// A list of captured images.
+  ///
+  /// Some backtracing algorithms may require this information, in which case
+  /// it will be filled in by the `capture()` method.  Other algorithms may
+  /// not, in which case it will be empty and you can capture an image list
+  /// separately yourself using `captureImages()`.
+  public var images: [Image]?
+
+  /// Holds information about the shared cache.
+  public struct SharedCacheInfo: Sendable {
+    #if os(macOS) || os(iOS) || os(watchOS) || os(tvOS)
```

Comment:
This is public API, right? Shouldn't we aim for API compat between the platforms and have these available elsewhere? Otherwise we force users into this ugly `#if os` stuff.

Reply:
Amusingly that's exactly what I thought about the `struct` itself, but I'd decided the fields were things that were only applicable to Darwin. That is, I wanted to make sure that anyone writing code against some other platform made certain to handle the `SharedCacheInfo`, but I didn't want to expose the fields there because they aren't meaningful.

I could make them `internal` I suppose, but the downside of that is that there's a chance on Darwin that you might want to display their contents from something outside of this code.

Reply:
Decided to make the fields accessible but never return an object of this type, which makes more things `Optional` elsewhere but will make the API the same everywhere.